ts/features/itwallet/walletInstance/store/selectors/index.ts

```diff
@@ -0,0 +1,34 @@
+import * as O from "fp-ts/lib/Option";
+import { flow, pipe } from "fp-ts/lib/function";
+import { createSelector } from "reselect";
+import { GlobalState } from "../../../../../store/reducers/types";
+import { isWalletInstanceAttestationValid } from "../../../common/utils/itwAttestationUtils";
+
+/* Selector to get the wallet instance attestation */
+export const itwWalletInstanceAttestationSelector = (state: GlobalState) =>
+  state.features.itWallet.walletInstance.attestation;
+
+/* Selector to check if the attestation is valid */
+export const itwIsWalletInstanceAttestationValidSelector = createSelector(
+  itwWalletInstanceAttestationSelector,
+  flow(
+    O.fromNullable,
+    O.map(isWalletInstanceAttestationValid),
+    O.getOrElse(() => false)
+  )
+);
+
+/* Selector to get the wallet instance status */
+export const itwWalletInstanceStatusSelector = createSelector(
+  (state: GlobalState) => state.features.itWallet.walletInstance,
+  walletInstance =>
+    pipe(
+      O.fromNullable(walletInstance.isRevoked),
+      O.filter(Boolean),
```

Comment:
Do we really need to filter here if we are also interested in `false` values?

```js
pipe(
  O.fromNullable(walletInstance.isRevoked),
  O.map((isRevoked) => ({
    isRevoked,
    revocationReason: walletInstance.revocationReason
  })),
  O.getOrElse(() => ({ isRevoked: false }))
```

Reply:
The filter was intentionally added to ensure that, when `isRevoked` is `false`, the result does not include a `revocationReason` property with an `undefined` value.
However, if you see this as unnecessary or have a better approach to achieve the same result, I’d be glad to consider any alternative ideas you might have